controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/ApplicationList.java

```diff
@@ -160,13 +160,13 @@ public ApplicationList canUpgradeAt(Instant instant) {
     }
 
     /**
-     * Returns the subset of applications that hasn't pinned to an an earlier major version than the given one.
+     * Returns the subset of applications that hasn't pinned to another major version than the given one.
      *
      * @param targetMajorVersion the target major version which applications returned allows upgrading to
      * @param defaultMajorVersion the default major version to assume for applications not specifying one
      */
     public ApplicationList allowMajorVersion(int targetMajorVersion, int defaultMajorVersion) {
-        return listOf(list.stream().filter(a -> a.deploymentSpec().majorVersion().orElse(defaultMajorVersion)
+        return listOf(list.stream().filter(a -> a.majorVersion().orElse(a.deploymentSpec().majorVersion().orElse(defaultMajorVersion))
                                            >= targetMajorVersion));
```

Comment:
Doc was in agreement with this, but not any more.

Reply:
I think it's still correct

Reply:
Right, I forgot I edited this comment independently of the semantic change. I'll correct it back.